queries.sql

```diff
@@ -4,96 +4,119 @@
+SELECT firstName, lastName, emailAddress, phone, profileURL, hobby, habitGoal
+    FROM UserTable, Buddies
+    WHERE (buddy1 = UserTable.ID
+        OR buddy2 = UserTable.ID)
 
--- SELECT 
---     FROM
-
---     ORDER BY 
+    ORDER BY lastName ASC
 
 --Edit Profile
 ---NEED TO READ: Current user data, firstName, lastName, habitGoal, habitCategory, hobby, email, profile url
 ---NEED TO WRITE: Everything from above
-SELECT *
-    FROM Username
-
-    -- ORDER BY 
+SELECT firstName, lastName, emailAddress, phone, profileURL, hobby, habitGoal, 
+    FROM UserTable
+    WHERE ID = ${req.params.ID}
+
+UPDATE UserTable
+    SET firstName=$(),                      --UPDATE
+        lastName=$(),                       --UPDATE
+        emailAddress=$(),                   --UPDATE
+        phone=$(),                          --UPDATE
+        username=$(username),
+        password=$(password),
+        dob=$(),                            --UPDATE
+        profileURL=$(),                     --UPDATE
+        hobby=$(),                          --UPDATE
+        habitGoal=$(),                      --UPDATE
+        notifications=$(notifications),
+        theme=$(theme)
+
+    WHERE ID = ${req.params.ID}
 
 --Empty Habits 
 ---NEED TO READ: None
 ---NEED TO WRITE: habits, written specifically to the userprofile
 
-SELECT 
-    FROM
+UPDATE UserTable
+    SET firstName=$(firstName),
+        lastName=$(lastName),
+        emailAddress=$(emailAddress),
+        phone=$(phone),
+        username=$(username),
+        password=$(password),
+        dob=$(dob),
+        profileURL=$(profileURL),
+        hobby=$(hobby),
+        habitGoal=$(),                      --UPDATE
+        notifications=$(notifications),
+        theme=$(theme)
+
+    WHERE ID = ${req.params.ID}
 
-    ORDER BY 
 
 --Empty Profile
 ---NEED TO READ: None
 ---NEED TO WRITE: Writing all user profile data. Similar to the edit profile. Current user data, firstName, lastName, habitGoal, habitCategory, hobby, email, profile url
 
-SELECT 
-    FROM
-
-    ORDER BY 
-
---Habit Selector
----NEED TO READ: user data: HabitCategory
----NEED TO WRITE: None
-SELECT 
-    FROM 
-
-    ORDER BY 
+UPDATE UserTable
+    SET firstName=$(),                      --UPDATE
+        lastName=$(),                       --UPDATE
+        emailAddress=$(),                   --UPDATE
+        phone=$(),                          --UPDATE
+        username=$(username),
+        password=$(password),
+        dob=$(),                            --UPDATE
+        profileURL=$(),                     --UPDATE
+        hobby=$(),                          --UPDATE
+        habitGoal=$(),                      --UPDATE
+        notifications=$(notifications),
+        theme=$(theme)
+
+    WHERE ID = ${req.params.ID}
 
 --Habit Tracker
 ---NEED TO READ: N/A
 ---NEED TO WRITE: N/A
 SELECT *
-    FROM Habit, UserHabit, HabitCategory, Username
-    WHERE userID = Username.ID
-        AND habitID = Habit.ID
+    FROM 
```

Comment:
Huh...how do you store the number of boxes that people have highlighted or whatever?

Reply:
I don't really know. Belina would be the person to talk to about that. I don't really know what she wants. Maybe an integer that stores the number of days, but I don't know.